File: mouse.py
import pyautogui
import time
from PIL import ImageGrab

def get_mouse_color():
    x, y = pyautogui.position()
    screenshot = ImageGrab.grab()
    color = screenshot.getpixel((x, y))

    return color


if __name__ == "__main__":
    time.sleep(1.5)
    pyautogui.click()
    current = get_mouse_color()
    
    while True:
        pixel_color = get_mouse_color()
        print(pixel_color[0])
        # if pixel color change > 4, then click
        if abs(pixel_color[0] - current[0]) > 4:
            pyautogui.click()
            break

File: mouse.c
#include <stdio.h>
#include <windows.h>
#include <stdlib.h>
#include <time.h>

COLORREF getMouseColor() {
    POINT p;
    GetCursorPos(&p);

    HDC hdc = GetDC(NULL);
    COLORREF color = GetPixel(hdc, p.x, p.y);
    ReleaseDC(NULL, hdc);

    return color;
}

int main() {
    Sleep(1500);
    mouse_event(MOUSEEVENTF_LEFTDOWN, 0, 0, 0, 0);
    mouse_event(MOUSEEVENTF_LEFTUP, 0, 0, 0, 0);

    COLORREF current = getMouseColor();
    COLORREF pixelColor;

    while (1) {
        pixelColor = getMouseColor();
        printf("%d\n", GetRValue(pixelColor));

        if (abs(GetRValue(pixelColor) - GetRValue(current)) > 4) {
            mouse_event(MOUSEEVENTF_LEFTDOWN, 0, 0, 0, 0);
            mouse_event(MOUSEEVENTF_LEFTUP, 0, 0, 0, 0);
            break;
        }
    }

    return 0;
}
